src/lib/viewers/box3d/model3d/Model3DViewer.js

```diff
@@ -239,15 +256,31 @@ class Model3DViewer extends Box3DViewer {
         if (animations.length > 0) {
             const clipIds = animations[0].getClipIds();
 
-            clipIds.forEach(clipId => {
-                const clip = animations[0].getClip(clipId);
-                const duration = clip.stop - clip.start;
-                this.controls.addAnimationClip(clipId, clip.name, duration);
-            });
-
-            if (clipIds.length > 0) {
-                this.controls.showAnimationControls();
-                this.controls.selectAnimationClip(clipIds[0]);
+            if (this.useReactControls) {
+                this.animationClips = clipIds.map(clipId => {
+                    const { name, start, stop } = animations[0].getClip(clipId);
+                    const duration = stop - start;
+                    return {
+                        duration,
+                        id: clipId,
+                        name,
+                    };
+                });
+
+                this.renderer.setAnimationClip(this.animationClips[0].id);
```

Comment:
Should we be acting on `renderer` inside of a controls-specific method?

Reply:
Ideally not, but it was being done previously in a more roundabout way -- L282 sets the animation clip => Model3DControls would emit a `EVENT_SELECT_ANIMATION_CLIP` event => `handleSelectAnimationClip` which acts on the `renderer`